src/H5Eint.c

```diff
@@ -862,34 +1676,61 @@ H5E__clear_entries(H5E_t *estack, size_t nentries)
 /*-------------------------------------------------------------------------
  * Function:    H5E_clear_stack
  *
- * Purpose:     Private function to clear the error stack for the
- *              specified error stack.
+ * Purpose:     Clear the default error stack
  *
  * Return:      SUCCEED/FAIL
  *
  *-------------------------------------------------------------------------
  */
 herr_t
-H5E_clear_stack(H5E_t *estack)
+H5E_clear_stack(void)
 {
-    herr_t ret_value = SUCCEED; /* Return value */
+    H5E_stack_t *estack;              /* Error stack to clear */
+    herr_t       ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_NOAPI(FAIL)
 
+    /* Get 'default' error stack */
+    if (NULL == (estack = H5E__get_my_stack()))
+        HGOTO_ERROR(H5E_ERROR, H5E_CANTGET, FAIL, "can't get current error stack");
+
+    /* Empty the error stack */
+    if (H5E__clear_entries(estack, estack->nused) < 0)
+        HGOTO_ERROR(H5E_ERROR, H5E_CANTSET, FAIL, "can't clear error stack");
+
+done:
+    FUNC_LEAVE_NOAPI(ret_value)
+} /* end H5E_clear_stack() */
+
+/*-------------------------------------------------------------------------
```

Comment:
This is analogous to H5E_clear_stack(), but is internal to the H5E package, to allow clearing any error stack.